From f00ddde81d8a4699b2bdd9b0dcca4bad54ce5e4f Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 19 Oct 2018 16:50:11 +0300
Subject: [PATCH] uboot: gaudi/goya: switch to use waitcmd command

The new waitcmd command has been introduced and it wraps the getfit
command now.
Switch to use the new command.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/gaudi/uboot-env-asic.txt b/board/habanalabs/gaudi/uboot-env-asic.txt
index c1050ab..c35886a 100644
--- a/board/habanalabs/gaudi/uboot-env-asic.txt
+++ b/board/habanalabs/gaudi/uboot-env-asic.txt
@@ -1,2 +1,2 @@
 bootm_low=0
-bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
+bootcmd=while true; do waitcmd; run command; done
diff --git a/board/habanalabs/gaudi/uboot-env-onix.txt b/board/habanalabs/gaudi/uboot-env-onix.txt
index c1050ab..c35886a 100644
--- a/board/habanalabs/gaudi/uboot-env-onix.txt
+++ b/board/habanalabs/gaudi/uboot-env-onix.txt
@@ -1,2 +1,2 @@
 bootm_low=0
-bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
+bootcmd=while true; do waitcmd; run command; done
diff --git a/board/habanalabs/gaudi/uboot-env-pldm.txt b/board/habanalabs/gaudi/uboot-env-pldm.txt
index 3fc2c10..c35886a 100644
--- a/board/habanalabs/gaudi/uboot-env-pldm.txt
+++ b/board/habanalabs/gaudi/uboot-env-pldm.txt
@@ -1,2 +1,2 @@
 bootm_low=0
-bootcmd=getfit && bootm $fileaddr#conf@1
+bootcmd=while true; do waitcmd; run command; done
diff --git a/board/habanalabs/gaudi/uboot_asic.fragment b/board/habanalabs/gaudi/uboot_asic.fragment
index 95c2b89..7fda35e 100644
--- a/board/habanalabs/gaudi/uboot_asic.fragment
+++ b/board/habanalabs/gaudi/uboot_asic.fragment
@@ -3,7 +3,7 @@
 CONFIG_BAUDRATE=115200
 CONFIG_BOOTDELAY=0
 CONFIG_DEBUG_UART_CLOCK=107500000
-CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
+CONFIG_BOOTCOMMAND="while true; do waitcmd; run command; done"
 CONFIG_DEFAULT_DEVICE_TREE="gaudi_asic"
 CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_SYS_I2C_EEPROM_BUS=1
diff --git a/board/habanalabs/gaudi/uboot_onix.fragment b/board/habanalabs/gaudi/uboot_onix.fragment
index bcf9e7a..32d1571 100644
--- a/board/habanalabs/gaudi/uboot_onix.fragment
+++ b/board/habanalabs/gaudi/uboot_onix.fragment
@@ -3,7 +3,7 @@
 CONFIG_BAUDRATE=57600
 CONFIG_BOOTDELAY=10
 CONFIG_DEBUG_UART_CLOCK=35000000
-CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
+CONFIG_BOOTCOMMAND="while true; do waitcmd; run command; done"
 CONFIG_DEFAULT_DEVICE_TREE="gaudi_onix"
 CONFIG_SPI_FLASH_STMICRO=y
 CONFIG_SPI_FLASH_MACRONIX=y
diff --git a/board/habanalabs/gaudi/uboot_pldm.fragment b/board/habanalabs/gaudi/uboot_pldm.fragment
index 2990a30..0c5974d 100644
--- a/board/habanalabs/gaudi/uboot_pldm.fragment
+++ b/board/habanalabs/gaudi/uboot_pldm.fragment
@@ -3,7 +3,7 @@
 CONFIG_BAUDRATE=1200
 CONFIG_BOOTDELAY=10
 CONFIG_DEBUG_UART_CLOCK=20200
-CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
+CONFIG_BOOTCOMMAND="while true; do waitcmd; run command; done"
 CONFIG_DEFAULT_DEVICE_TREE="gaudi_pldm"
 CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_PHY_FIXED=y
diff --git a/board/habanalabs/goya/uboot-env-asic.txt b/board/habanalabs/goya/uboot-env-asic.txt
index d0d5134..57b9192 100644
--- a/board/habanalabs/goya/uboot-env-asic.txt
+++ b/board/habanalabs/goya/uboot-env-asic.txt
@@ -1,4 +1,4 @@
 bootm_low=0
 ethaddr=00:33:66:99:cc:ff
 netboot=dhcp 192.168.10.109:goya-fit.itb && bootm $fileaddr#conf@1
-bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
+bootcmd=while true; do waitcmd; run command; done
diff --git a/board/habanalabs/goya/uboot-env-onix.txt b/board/habanalabs/goya/uboot-env-onix.txt
index f0d4e91..c823bf6 100644
--- a/board/habanalabs/goya/uboot-env-onix.txt
+++ b/board/habanalabs/goya/uboot-env-onix.txt
@@ -1,4 +1,4 @@
 bootm_low=0
 ethaddr=00:11:22:33:44:55
 netboot=dhcp 192.168.10.86:goya-fit.itb && bootm $fileaddr#conf@1
-bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
+bootcmd=while true; do waitcmd; run command; done
diff --git a/board/habanalabs/goya/uboot-env-pldm.txt b/board/habanalabs/goya/uboot-env-pldm.txt
index 9f5255b..7649fbf 100644
--- a/board/habanalabs/goya/uboot-env-pldm.txt
+++ b/board/habanalabs/goya/uboot-env-pldm.txt
@@ -1,3 +1,3 @@
 bootm_low=0
 ethaddr=00:22:44:66:88:aa
-bootcmd=getfit && bootm $fileaddr#conf@1
+bootcmd=while true; do waitcmd; run command; done
diff --git a/board/habanalabs/goya/uboot_asic.fragment b/board/habanalabs/goya/uboot_asic.fragment
index 83f2fc2..e5f5c4f 100644
--- a/board/habanalabs/goya/uboot_asic.fragment
+++ b/board/habanalabs/goya/uboot_asic.fragment
@@ -3,7 +3,7 @@
 CONFIG_BAUDRATE=115200
 CONFIG_BOOTDELAY=0
 CONFIG_DEBUG_UART_CLOCK=23888888
-CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
+CONFIG_BOOTCOMMAND="while true; do waitcmd; run command; done"
 CONFIG_DEFAULT_DEVICE_TREE="goya_asic"
 CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_SYS_I2C_EEPROM_BUS=1
diff --git a/board/habanalabs/goya/uboot_onix.fragment b/board/habanalabs/goya/uboot_onix.fragment
index 8dc7273..5b96d21 100644
--- a/board/habanalabs/goya/uboot_onix.fragment
+++ b/board/habanalabs/goya/uboot_onix.fragment
@@ -3,7 +3,7 @@
 CONFIG_BAUDRATE=57600
 CONFIG_BOOTDELAY=10
 CONFIG_DEBUG_UART_CLOCK=50000000
-CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
+CONFIG_BOOTCOMMAND="while true; do waitcmd; run command; done"
 CONFIG_DEFAULT_DEVICE_TREE="goya_onix"
 CONFIG_SPI_FLASH_STMICRO=y
 CONFIG_SPI_FLASH_MACRONIX=y
diff --git a/board/habanalabs/goya/uboot_pldm.fragment b/board/habanalabs/goya/uboot_pldm.fragment
index 007cd03..9e2d50c 100644
--- a/board/habanalabs/goya/uboot_pldm.fragment
+++ b/board/habanalabs/goya/uboot_pldm.fragment
@@ -3,7 +3,7 @@
 CONFIG_BAUDRATE=1200
 CONFIG_BOOTDELAY=10
 CONFIG_DEBUG_UART_CLOCK=25000
-CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
+CONFIG_BOOTCOMMAND="while true; do waitcmd; run command; done"
 CONFIG_DEFAULT_DEVICE_TREE="goya_pldm"
 CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_PHY_FIXED=y
